src/amo/components/ServerError/index.js

```diff
@@ -0,0 +1,71 @@
+import React, { PropTypes } from 'react';
+import { oneLine } from 'common-tags';
+import { compose } from 'redux';
+import NestedStatus from 'react-nested-status';
+
+import Link from 'amo/components/Link';
+import { ADDON_TYPE_EXTENSION, ADDON_TYPE_THEME } from 'core/constants';
+import translate from 'core/i18n/translate';
+import { sanitizeHTML, visibleAddonType } from 'core/utils';
+import Card from 'ui/components/Card';
+
+import 'amo/components/ErrorPage/ErrorPage.scss';
+
+
+export class ServerErrorBase extends React.Component {
+  static propTypes = {
+    i18n: PropTypes.object.isRequired,
+  }
+
+  render() {
+    const { i18n } = this.props;
+
+    const fileAnIssueText = i18n.gettext(oneLine`
+      If you have additional information that would help us you can
+      <a href="https://github.com/mozilla/addons-frontend/issues/new/">file an
+      issue</a>. Tell us what steps you took that lead to the error and we'll
+      do our best to fix it.`);
+
+    /* eslint-disable react/no-danger */
+    return (
+      <NestedStatus code={500}>
+        <Card className="ErrorPage ServerError"
+          header={i18n.gettext('Server Error')}>
+          <p>
+            {i18n.gettext(oneLine`
+              Sorry, but there was an error with our server and we couldn't
+              complete your request. We have logged this error and will
+              investigate it.`)}
+          </p>
+
+          <h2>{i18n.gettext('Suggested Pages')}</h2>
```

Comment:
Actually, this is a good idea but mind if I do it in my consolidating-these-errors-into-a-shared-component follow-up PR?